data_structures/src/radon_error.rs

```diff
@@ -0,0 +1,147 @@
+use std::convert::TryFrom;
+use std::io::Cursor;
+
+use cbor::types::Tag;
+use cbor::value::Value;
+use cbor::GenericEncoder;
+use num_enum::IntoPrimitive;
+
+#[derive(Clone, Copy, Debug, IntoPrimitive)]
+#[repr(u8)]
+/// List of RADON-level errors.
+/// **WARNING: these codes are consensus-critical.** They can be renamed but they cannot be
+/// re-assigned without causing a non-backwards-compatible protocol upgrade.
+pub enum RadonErrors {
+    /// Unknown error. Something went really bad!
+    Unknown = 0x00,
+    // Script format errors
+    /// At least one of the source scripts is not a valid CBOR-encoded value.
+    SourceScriptNotCBOR = 0x01,
```

Comment:
How about unifying this with `SourceScriptNotArray`?

```suggestion
    SourceScriptNotCBORArray = 0x01,
```